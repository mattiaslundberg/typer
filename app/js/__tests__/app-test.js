import React from 'react'
import App from '../app'
import sd from 'skin-deep'

describe('AppTest', () => {
  const render = () => {
    return sd.shallowRender(
      <App />
    )
  }

  it('Renders Typer with text', () => {
    let tree = render()

    expect(tree.getRenderOutput()).toMatchSnapshot()
  })
})
