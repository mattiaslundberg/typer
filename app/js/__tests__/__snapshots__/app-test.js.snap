exports[`AppTest Renders TextSelector 1`] = `
<div>
  <OAuth
    authCallback={[Function]} />
  <TextSelector
    isAuthenticated={false}
    onSelect={[Function]}
    options={Array []} />
</div>
`;

exports[`AppTest Renders TextSelector 2`] = `
<div>
  <OAuth
    authCallback={[Function]} />
  <TextSelector
    isAuthenticated={false}
    onSelect={[Function]}
    options={
      Array [
        Object {
          "name": "one",
          "value": "one",
        },
      ]
    } />
</div>
`;

exports[`AppTest Renders Typer after text is set 1`] = `
<div>
  <OAuth
    authCallback={[Function]} />
  <Typer
    fullText={
      Array [
        "H",
        "e",
        "l",
        "l",
        "o",
        " ",
        "w",
        "o",
        "r",
        "l",
        "d",
        "!",
      ]
    }
    onNewClick={[Function]} />
</div>
`;

exports[`AppTest Renders error view 1`] = `
<div>
  <OAuth
    authCallback={[Function]} />
  <div
    className="error-message">
    Something went wrong
  </div>
</div>
`;

exports[`AppTest Renders login link after getting the url 1`] = `
<div>
  <OAuth
    authCallback={[Function]} />
  <TextSelector
    isAuthenticated={false}
    onSelect={[Function]}
    options={Array []} />
</div>
`;
